src/gf.c

```diff
@@ -3360,6 +3396,13 @@ JL_DLLEXPORT jl_value_t *jl_apply_generic(jl_value_t *F, jl_value_t **args, uint
                                                      jl_int32hash_fast(jl_return_address()),
                                                      world);
     JL_GC_PROMISE_ROOTED(mfunc);
+    uint8_t miflags = jl_atomic_load_relaxed(&mfunc->flags);
```

Comment:
Are we sure this has no overhead? Could it go in a slow path?

Reply:
Good point, not sure. I think the cost should be negligible. Thoughts @vtjnash?

Could add an outer check for `jloptions.trace_dispatch`? Remove the check from inside the `record... ` function?

Reply:
I've moved this atomic load off the fast path.

Reply:
This is probably better but I think it should go in a slower lookup path in jl_lookup_generic; at least only when the fast cache misses.

Reply:
Ah, I see. Going by the comments in `jl_lookup_generic_`, there is a lookup in an associative cache, and if that misses, a lookup in the full cache. Can a method instance be in either of those caches if it hasn't been dispatched? If not, then I could move the call [here](https://github.com/JuliaLang/julia/blob/master/src/gf.c#L3337)?

Reply:
I've moved it [here](https://github.com/JuliaLang/julia/pull/55848/files#diff-9a718f382a240acdbe9d0bdbc3ffb521f67af0e62b2ba8c8594b8faa65c78014R3381-R3390); please take a look?